### PR TITLE
Add deleteInsurance command

### PR DESCRIPTION
**Pull request by matthewyip1511**
I have implemented the deleteInsurance command in a (very) similar fashion to the add insurance command. As the insurance object has not been fully implemented into the `Person`, I have likewise not added that feature in this PR but will do so later

**Note to reviewer** : as much of the code was copied from the addInsurance side, please help me double check that all names / variables have been changed correctly and no trails of addInsurance are still there. Also help me check that the right tests are implemented for this feature.

Thanks!

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/AddInsuranceCommand.java [hunk 1]
```diff
@@ -14,7 +14,7 @@ public class AddInsuranceCommand extends Command {
     public static final String COMMAND_WORD = "addInsurance";
     public static final String MESSAGE_USAGE = COMMAND_WORD
             + ": Adds an insurance plan to the client identified "
-            + "by their name. \n"
+            + "by their client id. \n"
```
Comment (matthewyip1511) [inline src/main/java/seedu/address/logic/commands/AddInsuranceCommand.java:17]: changing this as discussed

File: src/test/java/seedu/address/logic/parser/AddressBookParserTest.java [hunk 2]
```diff
@@ -104,11 +99,28 @@ public void parseCommand_addInsuranceCommand() throws Exception {
...
+                + INDEX_FIRST_PERSON.getOneBased()
+                + " "
+                + PREFIX_INSURANCE_ID
+                + " 0 ";
+        DeleteInsuranceCommand command = (DeleteInsuranceCommand) parser.parseCommand(userInput);
+        assertEquals(new DeleteInsuranceCommand(INDEX_FIRST_PERSON, 0), command);
+    }
+
+    @Test
+    public void parseCommand_unrecognisedInput_throwsParseException() {
```
Comment (matthewyip1511) [inline src/test/java/seedu/address/logic/parser/AddressBookParserTest.java:119]: re-ordered to move add and delete insurance above the tests for unrecognised inputs

File: src/main/java/seedu/address/logic/parser/DeleteInsuranceCommandParser.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+/**
+ * Parses input arguments and adds {@code DeleteInsurance} object
+ */
+public class DeleteInsuranceCommandParser implements Parser<DeleteInsuranceCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the {@code RemarkCommand}
+     * and returns a {@code AddInsuranceCommand} object for execution.
+     *
```
Comment (zi-yii) [inline src/main/java/seedu/address/logic/parser/DeleteInsuranceCommandParser.java:20]: JavaDoc has traces of AddInsuranceCommand and RemarkCommand

File: src/main/java/seedu/address/logic/parser/AddressBookParser.java [hunk 4]
```diff
@@ -81,6 +82,10 @@ public Command parseCommand(String userInput) throws ParseException {
         case AddInsuranceCommand.COMMAND_WORD:
             return new AddInsuranceCommandParser().parse(arguments);
 
+        case DeleteInsuranceCommand.COMMAND_WORD:
+            return new DeleteInsuranceCommandParser().parse(arguments);
+
+
```
Comment (zi-yii) [inline src/main/java/seedu/address/logic/parser/AddressBookParser.java:88]: To remove extra line

File: src/test/java/seedu/address/logic/parser/AddressBookParserTest.java (hunk 2 above)
Comment (zi-yii) [reply src/test/java/seedu/address/logic/parser/AddressBookParserTest.java:119]: Sure 👍

**Review by zi-yii — COMMENTED** [review]
LGTM

**Review by RezwanAhmed123 — APPROVED** [review]
LGTM, minor nits, will fix myself

**Comment by RezwanAhmed123** [comment]
link to #66

**Pull request merged**